sdk/src/Core/Amazon.Runtime/Credentials/_bcl45+netstandard/SSOAWSCredentials.cs

```diff
@@ -0,0 +1,344 @@
+        /// </summary>
+        public string RoleName { get; private set; }
+
+        /// <summary>
+        /// The main URL for users to login to their SSO directory.
+        /// Provided by the SSO service via the web console.
+        /// </summary>
+        public string StartUrl { get; private set; }
+
+        /// <summary>
+        /// Options to be used in the SSO flow to resolve credentials.
+        /// Developers wishing to support AWS SSO would want to provide the following:
+        /// - <seealso cref="SSOAWSCredentialsOptions.ClientName"/>
+        /// - <seealso cref="SSOAWSCredentialsOptions.SsoVerificationCallback"/>
+        /// </summary>
+        public SSOAWSCredentialsOptions Options { get; private set; }
+
+        /// <summary>
+        /// Constructs an SSOAWSCredentials object.
+        /// </summary>
+        /// <param name="accountId">The AWS account ID that temporary AWS credentials will be resolved for.</param>
+        /// <param name="region">The AWS region where the SSO directory for <paramref name="startUrl"/> is hosted.</param>
+        /// <param name="roleName">The corresponding IAM role in the AWS account that temporary AWS credentials will be resolved for.</param>
+        /// <param name="startUrl">The main URL for users to login to their SSO directory.</param>
+        public SSOAWSCredentials(string accountId, string region, string roleName, string startUrl)
+            : this(accountId, region, roleName, startUrl, new SSOAWSCredentialsOptions())
+        {
+        }
+
+        /// <summary>
+        /// Constructs an SSOAWSCredentials object.
+        /// </summary>
+        /// <param name="accountId">The AWS account ID that temporary AWS credentials will be resolved for.</param>
+        /// <param name="region">The AWS region where the SSO directory for <paramref name="startUrl"/> is hosted.</param>
+        /// <param name="roleName">The corresponding IAM role in the AWS account that temporary AWS credentials will be resolved for.</param>
+        /// <param name="startUrl">The main URL for users to login to their SSO directory.</param>
+        /// <param name="options">Options to be used in the SSO flow to resolve credentials.</param>
+        public SSOAWSCredentials(
+            string accountId, string region,
+            string roleName, string startUrl,
+            SSOAWSCredentialsOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            AccountId = accountId;
+            Region = region;
+            RoleName = roleName;
+            StartUrl = startUrl;
+            Options = options;
+        }
+
+#if BCL
+        protected override CredentialsRefreshState GenerateNewCredentials()
+        {
+            ValidateCredentialsInputs();
+
+            var oidcClient = CreateClient<ICoreAmazonSSOOIDC>(
```

Comment:
Should the service clients be member variables so they don't have to be recreated all of the time.

Reply:
I don't have enough insight on the SDK to make an appropriate call here. I did take inspiration from classes like `AssumeRoleAWSCredentials`, which creates an STS client within `GenerateNewCredentials`

Reply:
Okay, lets stay consistent for now.